api/handlers.go

```diff
@@ -1124,3 +1121,16 @@ func notImplementedHandler(c *context, w http.ResponseWriter, r *http.Request) {
 func optionsHandler(c *context, w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 }
+
+// Decode a base64-url-safe-encoded Registry Auth Config header.
+func decodeRegistryAuthHeader(r *http.Request, header string) []byte {
```

Comment:
This could technically decode any header that's given to it, right? Let's give it a more general name and hope for reuse.